flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/source/config/DynamodbStreamsSourceConfigConstants.java

```diff
@@ -40,9 +42,41 @@ public enum InitialPosition {
     public static final ConfigOption<Long> SHARD_DISCOVERY_INTERVAL_MILLIS =
             ConfigOptions.key("flink.shard.discovery.intervalmillis")
                     .longType()
-                    .defaultValue(10000L)
+                    .defaultValue(60000L)
                     .withDescription("The interval between each attempt to discover new shards.");
 
+    public static final ConfigOption<Integer> DESCRIBE_STREAM_INCONSISTENCY_RESOLUTION_RETRY_COUNT =
+            ConfigOptions.key("flink.describestream.inconsistencyresolution.retries")
+                    .intType()
+                    .defaultValue(5)
+                    .withDescription(
+                            "The number of times to retry build shard lineage if describestream returns inconsistent response");
+
+    public static final ConfigOption<Integer> DYNAMODB_STREAMS_RETRY_COUNT =
+            ConfigOptions.key("flink.dynamodbstreams.numretries")
+                    .intType()
+                    .defaultValue(50)
+                    .withDescription(
+                            "The number of times to retry DynamoDB Streams API call if it returns a retryable exception");
+
+    public static final ConfigOption<Duration> DYNAMODB_STREAMS_EXPONENTIAL_BACKOFF_MIN_DELAY =
+            ConfigOptions.key("flink.dynamodbstreams.backoff.mindelayduration")
+                    .durationType()
+                    .defaultValue(Duration.ofMillis(100))
+                    .withDescription(
+                            "The minimum delay for exponential backoff for describestream");
+
+    public static final ConfigOption<Duration> DYNAMODB_STREAMS_EXPONENTIAL_BACKOFF_MAX_DELAY =
+            ConfigOptions.key("flink.dynamodbstreams.backoff.maxdelay")
+                    .durationType()
+                    .defaultValue(Duration.ofMillis(1000))
+                    .withDescription(
+                            "The maximum delay for exponential backoff for describestream");
+
+    public static final Duration DDB_STREAMS_MAX_RETENTION_PERIOD = Duration.ofHours(24);
+
+    public static final String SHARD_ID_SEPARATOR = "-";
```

Comment:
These are not strictly `Config`, so putting it here in public makes it public interface. Let's make it private in the class that uses them

Reply:
Also ! I wonder if we should add 1 hour here to reduce race conditions?

Reply:
yeah, having 25 hours would reduce the race conditions. Will change it. Sure, will make these private in the class which uses them